src/Avalonia.Controls/Primitives/ToggleButton.cs

```diff
@@ -17,24 +23,80 @@ public class ToggleButton : Button
                 unsetValue: null,
                 defaultBindingMode: BindingMode.TwoWay);
 
+        /// <summary>
+        /// Defines the <see cref="IsThreeState"/> property.
+        /// </summary>
         public static readonly StyledProperty<bool> IsThreeStateProperty =
             AvaloniaProperty.Register<ToggleButton, bool>(nameof(IsThreeState));
 
+        /// <summary>
+        /// Defines the <see cref="Checked"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> CheckedEvent =
+            RoutedEvent.Register<ToggleButton, RoutedEventArgs>(nameof(Checked), RoutingStrategies.Bubble);
+
+        /// <summary>
+        /// Defines the <see cref="Unchecked"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> UncheckedEvent =
+            RoutedEvent.Register<ToggleButton, RoutedEventArgs>(nameof(Unchecked), RoutingStrategies.Bubble);
+
+        /// <summary>
+        /// Defines the <see cref="Unchecked"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> IndeterminateEvent =
+            RoutedEvent.Register<ToggleButton, RoutedEventArgs>(nameof(Indeterminate), RoutingStrategies.Bubble);
+
         private bool? _isChecked = false;
 
         static ToggleButton()
         {
             PseudoClass<ToggleButton, bool?>(IsCheckedProperty, c => c == true, ":checked");
             PseudoClass<ToggleButton, bool?>(IsCheckedProperty, c => c == false, ":unchecked");
             PseudoClass<ToggleButton, bool?>(IsCheckedProperty, c => c == null, ":indeterminate");
+
+            IsCheckedProperty.Changed.AddClassHandler<ToggleButton>((x, e) => x.OnIsCheckedChanged(e));
+        }
+
+        /// <summary>
+        /// Raised when a <see cref="ToggleButton"/> is checked.
+        /// </summary>
+        public event EventHandler<RoutedEventArgs> Checked
```

Comment:
Oh wow, WPF actually defines an event for every state of a `ToggleButton` huh? Not how I would have done it, but I'm :+1: on following WPF's API here.

Reply:
I needed something similar (even though it can be implemented using `IsChecked` property) at some point and I wondered what WPF does. UWP also defines events per state.